### PR TITLE
abseil-cpp: build shared

### PR DESCRIPTION
**Pull request by marsam**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117160` at f821b2a0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>42 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.arviz</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.rl-coach</li>
    <li>python38Packages.scikit-tda</li>
    <li>python38Packages.tensorflow-probability</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python39Packages.dm-sonnet</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/33aec8c93b997f25a7ce0020e2aa6dca">murmur</a></li>
    <li><a href="https://gist.github.com/3c3aa54d8d4a6310ab73ac4a9293bd00">python38Packages.cirq</a></li>
    <li><a href="https://gist.github.com/34b91df550f1a0c30e3f4c39ae6ee503">python39Packages.cirq</a></li>
  </ul>
</details>
<details>
  <summary>18 packages skipped due to time constraints:</summary>
  <ul>
    <li>kmymoney</li>
    <li>or-tools</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.labelbox</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.ortools</li>
    <li>python38Packages.tensorflow (python38Packages.tensorflow-build ,python38Packages.tensorflow-build_2 ,python38Packages.tensorflowWithoutCuda ,python38Packages.tensorflow_2)</li>
    <li>python38Packages.tflearn</li>
    <li>python38Packages.umap-learn</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>87 packages built successfully:</summary>
  <ul>
    <li>abseil-cpp</li>
    <li>beancount (python38Packages.beancount)</li>
    <li>bear</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duply</li>
    <li>dvc-with-remotes</li>
    <li>fava</li>
    <li>fcitx5-mozc</li>
    <li>gcalcli</li>
    <li>grpc</li>
    <li>ledger2beancount</li>
    <li>lieer</li>
    <li>python38Packages.beancount_docverif</li>
    <li>python38Packages.chirpstack-api</li>
    <li>python38Packages.goobook</li>
    <li>python38Packages.google-api-core</li>
    <li>python38Packages.google-api-python-client</li>
    <li>python38Packages.google-cloud-access-context-manager</li>
    <li>python38Packages.google-cloud-asset</li>
    <li>python38Packages.google-cloud-automl</li>
    <li>python38Packages.google-cloud-bigquery</li>
    <li>python38Packages.google-cloud-bigquery-datatransfer</li>
    <li>python38Packages.google-cloud-bigtable</li>
    <li>python38Packages.google-cloud-container</li>
    <li>python38Packages.google-cloud-core</li>
    <li>python38Packages.google-cloud-dataproc</li>
    <li>python38Packages.google-cloud-datastore</li>
    <li>python38Packages.google-cloud-dlp</li>
    <li>python38Packages.google-cloud-dns</li>
    <li>python38Packages.google-cloud-error-reporting</li>
    <li>python38Packages.google-cloud-firestore</li>
    <li>python38Packages.google-cloud-iam</li>
    <li>python38Packages.google-cloud-iot</li>
    <li>python38Packages.google-cloud-kms</li>
    <li>python38Packages.google-cloud-language</li>
    <li>python38Packages.google-cloud-logging</li>
    <li>python38Packages.google-cloud-monitoring</li>
    <li>python38Packages.google-cloud-org-policy</li>
    <li>python38Packages.google-cloud-os-config</li>
    <li>python38Packages.google-cloud-pubsub</li>
    <li>python38Packages.google-cloud-redis</li>
    <li>python38Packages.google-cloud-resource-manager</li>
    <li>python38Packages.google-cloud-runtimeconfig</li>
    <li>python38Packages.google-cloud-secret-manager</li>
    <li>python38Packages.google-cloud-securitycenter</li>
    <li>python38Packages.google-cloud-spanner</li>
    <li>python38Packages.google-cloud-speech</li>
    <li>python38Packages.google-cloud-storage</li>
    <li>python38Packages.google-cloud-tasks</li>
    <li>python38Packages.google-cloud-texttospeech</li>
    <li>python38Packages.google-cloud-trace</li>
    <li>python38Packages.google-cloud-translate</li>
    <li>python38Packages.google-cloud-videointelligence</li>
    <li>python38Packages.google-cloud-vision</li>
    <li>python38Packages.google-cloud-websecurityscanner</li>
    <li>python38Packages.googleapis-common-protos</li>
    <li>python38Packages.grpc_google_iam_v1</li>
    <li>python38Packages.oauthenticator</li>
    <li>python38Packages.proto-plus</li>
    <li>python38Packages.pydrive</li>
    <li>python38Packages.weboob</li>
    <li>python39Packages.beancount</li>
    <li>python39Packages.beancount_docverif</li>
    <li>python39Packages.chirpstack-api</li>
    <li>python39Packages.goobook</li>
    <li>python39Packages.google-api-core</li>
    <li>python39Packages.google-api-python-client</li>
    <li>python39Packages.google-cloud-access-context-manager</li>
    <li>python39Packages.google-cloud-bigtable</li>
    <li>python39Packages.google-cloud-core</li>
    <li>python39Packages.google-cloud-dns</li>
    <li>python39Packages.google-cloud-firestore</li>
    <li>python39Packages.google-cloud-logging</li>
    <li>python39Packages.google-cloud-org-policy</li>
    <li>python39Packages.google-cloud-resource-manager</li>
    <li>python39Packages.google-cloud-runtimeconfig</li>
    <li>python39Packages.google-cloud-trace</li>
    <li>python39Packages.google-cloud-videointelligence</li>
    <li>python39Packages.googleapis-common-protos</li>
    <li>python39Packages.grpc_google_iam_v1</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.proto-plus</li>
    <li>python39Packages.pydrive</li>
    <li>python39Packages.weboob</li>
    <li>qv2ray</li>
    <li>wal_e</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchgit`.
Near pkgs/tools/inputmethods/fcitx5/fcitx5-mozc.nix:1:60:
```
  |
1 | { lib, clangStdenv, fetchFromGitHub, fetchurl, fetchpatch, fetchgit
  |                                                            ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117160` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
43 packages marked as broken and skipped:
</summary>
<ul>
<li>linuxPackages-libre.sysdig</li>
<li>linuxPackages_4_4.sysdig</li>
<li>linuxPackages_4_9.sysdig</li>
<li>linuxPackages_hardkernel_4_14.sysdig</li>
<li>linuxPackages_hardkernel_latest.sysdig</li>
<li>linuxPackages_latest-libre.sysdig</li>
<li>python38Packages.arviz</li>
<li>python38Packages.dm-sonnet</li>
<li>python38Packages.graph_nets</li>
<li>python38Packages.optuna</li>
<li>python38Packages.pymc3</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.dm-sonnet</li>
<li>python39Packages.google-cloud-asset</li>
<li>python39Packages.google-cloud-automl</li>
<li>python39Packages.google-cloud-bigquery</li>
<li>python39Packages.google-cloud-bigquery-datatransfer</li>
<li>python39Packages.google-cloud-container</li>
<li>python39Packages.google-cloud-dataproc</li>
<li>python39Packages.google-cloud-datastore</li>
<li>python39Packages.google-cloud-dlp</li>
<li>python39Packages.google-cloud-error-reporting</li>
<li>python39Packages.google-cloud-iam</li>
<li>python39Packages.google-cloud-iot</li>
<li>python39Packages.google-cloud-kms</li>
<li>python39Packages.google-cloud-language</li>
<li>python39Packages.google-cloud-monitoring</li>
<li>python39Packages.google-cloud-os-config</li>
<li>python39Packages.google-cloud-pubsub</li>
<li>python39Packages.google-cloud-redis</li>
<li>python39Packages.google-cloud-secret-manager</li>
<li>python39Packages.google-cloud-securitycenter</li>
<li>python39Packages.google-cloud-spanner</li>
<li>python39Packages.google-cloud-speech</li>
<li>python39Packages.google-cloud-storage</li>
<li>python39Packages.google-cloud-tasks</li>
<li>python39Packages.google-cloud-texttospeech</li>
<li>python39Packages.google-cloud-translate</li>
<li>python39Packages.google-cloud-vision</li>
<li>python39Packages.google-cloud-websecurityscanner</li>
<li>python39Packages.graph_nets</li>
<li>python39Packages.optuna</li>
<li>python39Packages.rl-coach</li>
</ul>
</details>
<details>
<summary>
114 packages built:
</summary>
<ul>
<li>abseil-cpp</li>
<li>beancount (python38Packages.beancount)</li>
<li>bear</li>
<li>deja-dup</li>
<li>duplicity</li>
<li>duply</li>
<li>dvc-with-remotes</li>
<li>fava</li>
<li>fcitx5-mozc</li>
<li>gcalcli</li>
<li>google-cloud-cpp</li>
<li>grpc</li>
<li>kmymoney</li>
<li>ledger2beancount</li>
<li>lieer</li>
<li>linuxPackages.sysdig (linuxPackages_5_4.sysdig)</li>
<li>linuxPackages_4_14.sysdig</li>
<li>linuxPackages_4_19.sysdig</li>
<li>linuxPackages_5_10.sysdig</li>
<li>linuxPackages_5_11.sysdig (linuxPackages_latest.sysdig)</li>
<li>linuxPackages_hardened.sysdig</li>
<li>linuxPackages_latest_hardened.sysdig</li>
<li>linuxPackages_latest_xen_dom0.sysdig</li>
<li>linuxPackages_latest_xen_dom0_hardened.sysdig</li>
<li>linuxPackages_lqx.sysdig</li>
<li>linuxPackages_testing_bcachefs.sysdig</li>
<li>linuxPackages_xen_dom0.sysdig</li>
<li>linuxPackages_xen_dom0_hardened.sysdig</li>
<li>linuxPackages_zen.sysdig</li>
<li>murmur</li>
<li>or-tools</li>
<li>python38Packages.beancount_docverif</li>
<li>python38Packages.chirpstack-api</li>
<li>python38Packages.cirq</li>
<li>python38Packages.goobook</li>
<li>python38Packages.google-api-core</li>
<li>python38Packages.google-api-python-client</li>
<li>python38Packages.google-cloud-access-context-manager</li>
<li>python38Packages.google-cloud-asset</li>
<li>python38Packages.google-cloud-automl</li>
<li>python38Packages.google-cloud-bigquery</li>
<li>python38Packages.google-cloud-bigquery-datatransfer</li>
<li>python38Packages.google-cloud-bigtable</li>
<li>python38Packages.google-cloud-container</li>
<li>python38Packages.google-cloud-core</li>
<li>python38Packages.google-cloud-dataproc</li>
<li>python38Packages.google-cloud-datastore</li>
<li>python38Packages.google-cloud-dlp</li>
<li>python38Packages.google-cloud-dns</li>
<li>python38Packages.google-cloud-error-reporting</li>
<li>python38Packages.google-cloud-firestore</li>
<li>python38Packages.google-cloud-iam</li>
<li>python38Packages.google-cloud-iot</li>
<li>python38Packages.google-cloud-kms</li>
<li>python38Packages.google-cloud-language</li>
<li>python38Packages.google-cloud-logging</li>
<li>python38Packages.google-cloud-monitoring</li>
<li>python38Packages.google-cloud-org-policy</li>
<li>python38Packages.google-cloud-os-config</li>
<li>python38Packages.google-cloud-pubsub</li>
<li>python38Packages.google-cloud-redis</li>
<li>python38Packages.google-cloud-resource-manager</li>
<li>python38Packages.google-cloud-runtimeconfig</li>
<li>python38Packages.google-cloud-secret-manager</li>
<li>python38Packages.google-cloud-securitycenter</li>
<li>python38Packages.google-cloud-spanner</li>
<li>python38Packages.google-cloud-speech</li>
<li>python38Packages.google-cloud-storage</li>
<li>python38Packages.google-cloud-tasks</li>
<li>python38Packages.google-cloud-texttospeech</li>
<li>python38Packages.google-cloud-trace</li>
<li>python38Packages.google-cloud-translate</li>
<li>python38Packages.google-cloud-videointelligence</li>
<li>python38Packages.google-cloud-vision</li>
<li>python38Packages.google-cloud-websecurityscanner</li>
<li>python38Packages.googleapis-common-protos</li>
<li>python38Packages.grpc_google_iam_v1</li>
<li>python38Packages.labelbox</li>
<li>python38Packages.oauthenticator</li>
<li>python38Packages.ortools</li>
<li>python38Packages.proto-plus</li>
<li>python38Packages.pydrive</li>
<li>python38Packages.weboob</li>
<li>python39Packages.beancount</li>
<li>python39Packages.beancount_docverif</li>
<li>python39Packages.chirpstack-api</li>
<li>python39Packages.cirq</li>
<li>python39Packages.goobook</li>
<li>python39Packages.google-api-core</li>
<li>python39Packages.google-api-python-client</li>
<li>python39Packages.google-cloud-access-context-manager</li>
<li>python39Packages.google-cloud-bigtable</li>
<li>python39Packages.google-cloud-core</li>
<li>python39Packages.google-cloud-dns</li>
<li>python39Packages.google-cloud-firestore</li>
<li>python39Packages.google-cloud-logging</li>
<li>python39Packages.google-cloud-org-policy</li>
<li>python39Packages.google-cloud-resource-manager</li>
<li>python39Packages.google-cloud-runtimeconfig</li>
<li>python39Packages.google-cloud-trace</li>
<li>python39Packages.google-cloud-videointelligence</li>
<li>python39Packages.googleapis-common-protos</li>
<li>python39Packages.grpc_google_iam_v1</li>
<li>python39Packages.labelbox</li>
<li>python39Packages.oauthenticator</li>
<li>python39Packages.ortools</li>
<li>python39Packages.proto-plus</li>
<li>python39Packages.pydrive</li>
<li>python39Packages.weboob</li>
<li>qv2ray</li>
<li>rippled</li>
<li>rippled-validator-keys-tool</li>
<li>sysdig</li>
<li>wal_e</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Comment by rmcgibbo** [comment]
The `murmur` failure looks like it might be real. Was it fixed by the force push?

**Comment by marsam** [comment]
>The murmur failure looks like it might be real. Was it fixed by the force push?

yes, it's fixed

@GrahamcOfBorg build murmur
